pkg/controller/stack/stack_controller.go

```diff
@@ -266,18 +267,18 @@ func (r *ReconcileStack) Reconcile(ctx context.Context, request reconcile.Reques
 
 	// Step 1. Set up the workdir, select the right stack and populate config if supplied.
 
-	// Check which kind of source we have. Since GitSource is not "omitempty", so its fields can be
-	// inline, it will be an empty value rather than nil when not supplied.
+	// Check which kind of source we have.
 	if stack.GitSource != nil {
```

Comment:
Do we explicitly check if both `GitSource` and `FluxSource` are set? I think that would be a validation error, but right now it looks like the operator would silently choose the `GitSource` and ignore the `FluxSource`.

Reply:
Yeahp, good call. This would look better as a table-like switch, wouldn't it -- I should have thought ahead a bit when I did #332. I'll rewrite this bit.

Reply:
It'd be nice to have pattern matching and be able to do something like

```go
switch spec.GitSource, spec.FluxSource {
case gitSource, nil:
  // ...
case nil, fluxSource:
  // ...
default:
 // invalid!
}
```

but Go doesn't help us out much here. So instead, it's

```go
switch {
case gitSource != nil && fluxSource == nil:
 // etc.
}
```